charts/tekton-pipelines/templates/common/tasks/kustomize.yaml

```diff
@@ -0,0 +1,77 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: kustomize
+  namespace: {{ .Release.Namespace }}
+spec:
+  description: >-
+    Updates image tag in the kustomize overlay of the app
+
+  resources:
+    inputs:
+      - name: kubernetes-repo
+        type: git
+
+  params:
+    - name: application
+      type: string
+      description: name of the argocd application we're going to deploy/sync
+    
+    - name: image
+      type: string
+      description: new image for the application
+    
+    - name: kustomize_overlay_path
+      type: string
+      description: overlay path for kustomize call
+    
+    - name: kubernetes_branch
+      default: "main"
+    
+    - name: environment
+      type: string
+      description: environment name of the app being built, i.e. dev/staging/prod
+  
+  stepTemplate:
+    envFrom:
+      - configMapRef:
+          name: $(params.application)-build-pipeline-config
+
+  steps:
+    - name: git-set-user
+      image: {{ .Values.images.git | default "alpine/git:latest" }}
+      imagePullPolicy: {{ .Values.imagePullPolicy }}
+      workingDir: $(resources.inputs.kubernetes-repo.path)      
+      script: |
+        git config user.email "$DEVOPS_GROUP_EMAIL"
+        git config user.name "tekton-kustomize"
+
+    - name: update-image
+      image: {{ .Values.images.kustomize | default "k8s.gcr.io/kustomize/kustomize:latest"}}
+      imagePullPolicy: {{ .Values.imagePullPolicy }}
+      workingDir: $(resources.inputs.kubernetes-repo.path)
+      script: |
+        app=$(params.application)
+        env=$(params.environment)
+        application=${app%-$env}
+        cd $(params.kustomize_overlay_path) && \
+          /app/kustomize edit set image $application=$(params.image)
+    
+    - name: git-push
+      image: {{ .Values.images.git | default "alpine/git:latest" }}
+      imagePullPolicy: {{ .Values.imagePullPolicy }}
+      workingDir: $(resources.inputs.kubernetes-repo.path)      
+      script: |    
+        set +x
```

Comment:
I have new changes in this PR - https://github.com/saritasa-nest/saritasa-devops-tekton/pull/24

Reply:
Synced.